apps/system/js/sync/places_adapter.js

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* globals appWindowManager */
+/* exported PlacesAdapter */
```

Comment:
This code should end up being part of HistoryAdapter from the Sync App. The whole point of HistoryAdapter is to transform the data from Syncto into Places DS data and viceversa. We are spreading that logic across scripts (and even worse, across apps).